ring_buffer_it_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+)
+
+func TestRingbuffer_Add(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		sequence, err := rb.Add(context.Background(), "foo", hz.OverflowPolicyOverwrite)
+		assert.NoError(t, err)
+		assert.Equal(t, int64(0), sequence)
+
+		actualItem, err := rb.ReadOne(context.Background(), 0)
+		assert.NoError(t, err)
+		assert.Equal(t, "foo", actualItem)
+	})
+}
+
+func TestRingbuffer_AddNilElement(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		_, err := rb.Add(context.Background(), nil, hz.OverflowPolicyFail)
+		assert.Error(t, err)
+	})
+}
+
+func TestRingbuffer_AddAll(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		_, err := rb.AddAll(context.Background(), hz.OverflowPolicyOverwrite, "foo", "bar")
+		assert.NoError(t, err)
+
+		foo, err := rb.ReadOne(context.Background(), 0)
```

Comment:
Can you add tailSequence related lines here as well?

```java
public void addAll() throws Exception {
        Future<Long> f = clientRingbuffer.addAllAsync(asList("foo", "bar"), OVERWRITE).toCompletableFuture();
        Long result = f.get();

        assertEquals(new Long(serverRingbuffer.tailSequence()), result);
        assertEquals("foo", serverRingbuffer.readOne(0));
        assertEquals("bar", serverRingbuffer.readOne(1));
        assertEquals(0, serverRingbuffer.headSequence());
        assertEquals(1, serverRingbuffer.tailSequence());
    }
```

Reply:
Done. See f82945d3a0b32b6cf18f016f0fec68e5bf368675